components/quick_input/quick_input.tsx

```diff
@@ -11,65 +11,75 @@ import OverlayTrigger from 'components/overlay_trigger';
 import AutosizeTextarea from 'components/autosize_textarea';
 import Constants from 'utils/constants.jsx';
 
+export type Props = {
+
+    /**
+     * Whether to delay updating the value of the textbox from props. Should only be used
+     * on textboxes that to properly compose CJK characters as the user types.
+     */
+    delayInputUpdate?: boolean;
+
+    /**
+     * An optional React component that will be used instead of an HTML input when rendering
+     */
+    inputComponent?: ReactComponentLike;
```

Comment:
Same comment here and below wrt. `React.ReactNode` vs. `prop-types`

Reply:
Unfortunately, I tried to use ReactNode or other classes under React, but it seems that React.createElement line 180 does not accept it. Any idea?

Reply:
Would `React.ElementType` be a good solution?

Reply:
Ahh yeah it's not something we're rendering directly to the DOM. Maybe try `React.ComponentClass` or `React.Component`?

Reply:
I tried a lot of different combinations as suggested, `React.ElementType`, `React.Component`, `React.ComponentClass`... The latest does almost the job but fails line 183/184 because of `ref` and `default`, and in some other files because `inputComponent` is assigned to other types (cf. `marketplace_modal.tsx:201`, `edit_category_modal.tsx:133`). I'm out of solution to replace this legacy type, my level in Typescript reaches its limit, sorry!

Reply:
All good, let's stick with what you have then :) thanks for trying!